pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -330,8 +249,46 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 					}
 				}
 			}
+			var prCursor *string
+			var pullRequests []PullRequestNode
 
-			ghs.getRepoPullRequestInformation(repoInfo)
+			prOpenCount, err := getPullRequestCount(ctx, genClient, name, ghs.cfg.GitHubOrg, []PullRequestState{PullRequestStateOpen})
+			if err != nil {
+				ghs.logger.Sugar().Errorf("error getting open pull request count", zap.Error(err))
+			}
+			ghs.logger.Sugar().Debugf("open pull request count: %v for repo %v", prOpenCount, repo)
+
+			ghs.mb.RecordGitRepositoryPullRequestCountDataPoint(now, int64(prOpenCount.Repository.PullRequests.TotalCount), name)
+
+			prMergedCount, err := getPullRequestCount(ctx, genClient, name, ghs.cfg.GitHubOrg, []PullRequestState{PullRequestStateMerged})
+			if err != nil {
+				ghs.logger.Sugar().Errorf("error getting merged pull request count", zap.Error(err))
+			}
+
+			prPages := getNumPages(float64(100), float64(prOpenCount.Repository.PullRequests.TotalCount+prMergedCount.Repository.PullRequests.TotalCount))
```

Comment:
this should be a simple add function that we can unit test to ensure the output of the function is always going to be what we expect.